Egor Bogachick/Lesson19/Lesson19.Homeork/Program.cs

```diff
@@ -0,0 +1,248 @@
+
+            //foreach (var number in numbers2)
+            //{
+            //    Console.WriteLine($"{number.Number}: {number.Customer.Name}, {number.Customer.Birthday.Day}/{number.Customer.Birthday.Month}/{number.Customer.Birthday.Year}");
+            //}
+
+            #endregion
+
+            #endregion
+
+            #region WHERE
+            #region Завдання 1: Вивести інформацією про клієнтів, хто народився у січні
+            ////var clients = from c in customers
+            ////              where c.Birthday.Month == 1
+            ////              select c;
+
+            //var clients = customers.Where(c => c.Birthday.Month == 1);
+
+            //foreach (var client in clients)
+            //{
+            //    Console.Write(client.Name + " " + client.Birthday + " ");
+            //    foreach (var phone in client.PhoneNumbers)
+            //    {
+            //        Console.Write(phone + " ");
+            //    }
+            //Console.WriteLine();
+            //}
+
+            #endregion
+
+            #region Завдання 2: Вивести інформацією про клієнтів, імена яких мають непарну довжину
+            ////var clients = from c in customers
+            ////             where c.Name.Length % 2 == 1
+            ////             select c;
+
+            //var clients = customers.Where(c => c.Name.Length % 2 == 1);
+
+            //foreach (var client in clients)
+            //{
+            //    Console.Write(client.Name + " " + client.Birthday + " ");
+            //    foreach (var phone in client.PhoneNumbers)
+            //    {
+            //        Console.Write(phone + " ");
+            //    }
+            //    Console.WriteLine();
+            //}
+            #endregion
+
+            #region Завдання 3: Вивести інформацією про клієнтів, хто народилися у парному місяці року та хоча б один номер яких починається з "671"
+            //var clients = from c in customers
+            //              where c.Birthday.Month % 2 == 0 && c.PhoneNumbers.Any(x => x.StartsWith("671"))
+            //              select c;
+
+            //var clients = customers.Where(c => c.Birthday.Month % 2 == 0 && c.PhoneNumbers.Any(x => x.StartsWith("671")));
+
+            //foreach (var client in clients)
+            //{
+            //    Console.Write(client.Name + " " + client.Birthday + " ");
+            //    foreach (var phone in client.PhoneNumbers)
+            //    {
+            //        Console.Write(phone + " ");
+            //    }
+            //Console.WriteLine();
+            //}
+
+            #endregion
+
+            #endregion
+
+
+            var persons = JsonConvert.DeserializeObject<IEnumerable<Person>>(File.ReadAllText("data.json"));
+
+            #region third task
+
+            #region find out who is located farthest north/south/west/east using latitude/longitude data
+            //var north = persons.Max(x => x.Latitude);
```

Comment:
Also an option is to do .OrderByDescending(x=>x.Latitude).First() to not search for an entity after in loop